src/data/GraphOnChip.ts

```diff
@@ -757,22 +758,93 @@ export default class GraphOnChip {
         return [...nodes];
     }
 
-    getAllLinks(): NetworkLink[] {
-        const links: NetworkLink[] = [];
+    getAllLinksInitialState() {
+        const links: Record<
+            string,
+            {
+                chipId: number;
+                links: Record<string, LinkState>;
```

Comment:
same as above, it shoudl be obvious what the key is and its especially important in a nested record structure